### ISSUE TITLE
cilium_identity_count metric is buggy

### ISSUE THREAD

**Issue by joulaud** [cilium/cilium#13302]
## Bug report

**General Information**

- Cilium version `1.6.10 4df8eeb57 2020-07-01T21:54:22-07:00 go version go1.12.17 linux/amd64`
- Kernel version `SMP Debian 5.5.17-1~bpo10+1 (2020-04-23) x86_64 x86_64 x86_64 GNU/Linux`
- Orchestration system version in use: kubernetes`version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.9", GitCommit:"4fb7ed12476d57b8437ada90b4f93b17ffaeed99", GitTreeState:"clean", BuildDate:"2020-07-15T16:10:45Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}`

**How to reproduce the issue**

On a cilium-agent with metrics enabled we see that cilium_identity_count is completely unmeaningful (it should give the number of identities on the cluster but give some incoherent number, mostly negative numbers).

```
# HELP cilium_identity_count Number of identities currently allocated
# TYPE cilium_identity_count gauge
cilium_identity_count -5378
```

**Comment by joulaud** [comment]
cf. https://cilium.slack.com/archives/C53TG4J4R/p1601232078020500?thread_ts=1601203575.018300&cid=C53TG4J4R

**Comment by joestringer** [comment]
Thanks for the report! This is likely a sign of disbalanced counting on the specific metric - perhaps there's a missing increment, or multiple decrements for some path. Here's how to find the relevant metric:

```
$ git grep IdentityCount
daemon/cmd/daemon.go:           metrics.IdentityCount.Inc()
daemon/cmd/daemon.go:           metrics.IdentityCount.Add(float64(num))
pkg/identity/cache/allocator.go:                                metrics.IdentityCount.Inc()
pkg/identity/cache/allocator.go:                        metrics.IdentityCount.Dec()
pkg/metrics/metrics.go: // IdentityCount is the number of identities currently in use on the node
pkg/metrics/metrics.go: IdentityCount = NoOpGauge
pkg/metrics/metrics.go: IdentityCountEnabled                    bool
pkg/metrics/metrics.go:                 c.IdentityCountEnabled = true
pkg/metrics/metrics.go:                 IdentityCount = prometheus.NewGauge(prometheus.GaugeOpts{
pkg/metrics/metrics.go:                 collectors = append(collectors, IdentityCount)
pkg/metrics/metrics.go:                 c.IdentityCountEnabled = true
```

**Comment by aanm** [comment]
This is already fixed in 1.8 >= https://github.com/cilium/cilium/pull/12313
